### DISCUSSION TITLE
Request: turn on speakers when music starts playing

### DISCUSSION THREAD

**Discussion by ghost** [hassio-addons/addon-spotify-connect#86] [Ideas]
Feature Request:
Is it possible to add a feature to turn on a switch via home assistant when music starts streaming and turn it off again.
The speakers don't need to be on at all times, and turning them off via a switch would be very useful.

Happy to help contribute, just not sure where to start.

**Comment by ghost** [comment]
perhaps it could be extending spotifyd and using a hook like [on_song_change_hook ](https://github.com/Spotifyd/spotifyd/blob/master/docs/src/config/File.md)

**Comment by ghost** [comment]
it seems spotifyd has an [onevent type hook](https://github.com/Spotifyd/spotifyd/blob/dd2c203df01afff5577583155d448e6bf17578ae/src/player_event_handler.rs#L15). It was used [here ](https://github.com/Spotifyd/spotifyd/wiki/User-supplied-scripts) with a script to fetch metadata but I don't see why this couldn't fire a service call to home assistant.

**Comment by pax0r** [comment]
I have created my own fork with this hook, works like a charm:
https://github.com/pax0r/addon-spotify-connect/tree/master/spotify

Hass events are
- hassio_spotify_start
- hassio_spotify_stop
- hassio_spotify_change

**Reply by ghost** [reply]
That's awesome. I'll test it out today!

**Reply by giannello** [reply]
the same feature is also available in [my fork](https://github.com/giannello/addon-spotify-connect) - enable the relevant config option and it will expose an entity with the status of librespot

**Reply by frenck** [reply]
An external program should not create or maintain entities in Home Assistant like that.

See also: <https://www.home-assistant.io/blog/2021/05/12/integrations-api/>

**Reply by DaAwesomeP** [reply]
@frenck sending an event is the same way the NUT addon works. https://github.com/hassio-addons/addon-nut/blob/main/nut/rootfs/usr/bin/notify

I have implemented events in this way with librespot in my fork as well:
https://github.com/DaAwesomeP/ha-addon-spotify-connect-advanced/blob/main/spotify/rootfs/usr/bin/onevent
https://github.com/DaAwesomeP/ha-addon-spotify-connect-advanced/blob/main/spotify/rootfs/etc/services.d/spotifyd/run

**Reply by frenck** [reply]
> @frenck sending an event is the same way the NUT addon works. hassio-addons/addon-nut@main/nut/rootfs/usr/bin/notify

Yes, which is a mistake just as well.

> I have implemented events in this way with librespot in my fork as well:

You should not, see above comment and link.
